src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.h

```diff
@@ -83,6 +83,9 @@ struct OMapInnerNode
 	return;
       }
       set_seen_by_users();
+      // prior's "begin" and "end" are initialized when seen by users
+      // for the first time, so we need to initialize them here
+      init_range(prior.get_begin(), prior.get_end());
```

Comment:
How is it possible that a rewrite omap extent's range won't be initiated upon committing here ?

See do_on_rewrite() -> this->init_range(), it should be set already upon pending rewrite.

Reply:
Ah right!

Is this a more complete fix below?

In `do_on_rewrite(t, extent)`:
1. use `prior instance` instead of `extent`
2. initialize `this` based on whether `this` is seen (should be identical to whether `prior` is seen)
3. if not seen (else) add comments to explain that `prior` might become seen anytime by concurrent transactions, so we need to fix it upon `prepare_commit()` as well.
4. if not seen (else) add comments that it might also become seen later through `omap_load_extent()` in this transaction, might cause `prior` not seen but `this` is seen (?).

In `prepare_commit()`, are `this->parent_node_t::on_rewrite(t, ext);` and `this->sync_children_capacity();` also related for 3. ?